pandas/core/internals/concat.py

```diff
@@ -200,6 +202,21 @@ def concatenate_managers(
     if concat_axis == 0:
         return _concat_managers_axis0(mgrs_indexers, axes, copy)
 
+    if len(mgrs_indexers) > 0 and mgrs_indexers[0][0].nblocks > 0:
+        first_dtype = mgrs_indexers[0][0].blocks[0].dtype
```

Comment:
Any change to do an upcast if we have one/some float64 and one/some float32? That would generalize this pfastpath to covers all floats and not make a distinction between float32/float64.

Reply:
that would change behavior in some cases